linchpin/hooks/__init__.py

```diff
@@ -0,0 +1,150 @@
+"""
```

Comment:
Are hooks really not part of the API? Seems they might fit better there. Thoughts?